test/integration/TestFormat.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Test the behavior of 'swarm format' on selected input files.
+module TestFormat (testFormatting) where
+
+import Data.Text qualified as T
+import Swarm.Language.Format
+import Swarm.Language.Parser.Core (LanguageVersion (SwarmLangLatest))
+import Swarm.Util (findAllWithExt, readFileMayT)
+import Test.Tasty
+import Test.Tasty.HUnit
+import Witch (into)
+
+-- | Generate test tree to check that 'swarm format' is the identity
+--   on .sw files in @data/test/language-snippets/format@.
+testFormatting :: IO TestTree
+testFormatting = do
+  testFilePaths <- findAllWithExt "data/test/language-snippets/format" "sw"
+  pure . testGroup "Check code formatting" $ map checkFormat testFilePaths
+ where
+  defaultFormatCfg = FormatConfig Stdin Stdout Nothing SwarmLangLatest
+
+  checkFormat :: FilePath -> TestTree
+  checkFormat f = testCase f $ do
+    mcontent <- readFileMayT f
+    case mcontent of
+      Nothing -> assertFailure $ "Failed to load " <> f
+      Just content -> case formatSwarm defaultFormatCfg content of
+        Left err -> assertFailure (into @String err)
+        -- Make sure they are the same, but ignore leading + trailing whitespace
```

Comment:
Why ignore leading and trailing whitespace? Will the files not be the same?

Reply:
I don't remember exactly.  I think I just didn't want to be bothered getting such things exactly right, e.g. whether there should be a trailing newline and so on.  But we could try to be more precise about it if you think it would be worth it.

Reply:
I think it would be better to test precisely what we expect the formatter to do.

If this gets annoying on Windows, I would be okay with the tests ignoring the trailing newline, but I would like to know that the formatter does not add multiple lines.

Reply:
Oh, now I remember why, because my editor adds trailing newlines by default, but the formatter does not produce trailing newlines.  Creating files with no trailing newlines would just be annoying.  I will change it so it tests up to a single trailing newline but otherwise requires the whitespace to be exact.